WORKSPACE

```diff
@@ -80,3 +80,6 @@ filegroup(
 )
 """
 )
+
+load("//scala:scala.bzl", "register_scala_toolchains")
+register_scala_toolchains()
```

Comment:
@katre is there a way to not mandate this call by every one of our users? like a default toolchain

Reply:
Unfortunately, no. Bazel needs to know the available toolchains, WORKSPACE is the place for this to happen.

Eventually, when fully recursive workspace loading exists, you could place the register_scala_toolchains() call inside the WORKSPACE for the rules_scala repository, and that would be loaded, but for now it has to be in the root WORKSPACE.

Reply:
I see.
Two questions:
1. java_toolchain doesn't mandate this because it's on a different implementation, right?
2. once such a toolchain is declared can more properties be added with default values so that people won't break?

Reply:
1. Correct, java_toolchain doesn't use this mechanism (I'm adding support this month!) Once it does, java toolchains will need to be registered, with at least one being registered by default by Bazel's initialization system.

2. Yes, you can add further properties to the scala_toolchain. You can either include a default value in the scala_toolchain rule definition (possibly with a warning), or have the rules handle missing properties, whichever makes sense for your rule implementation.